test/libevmasm/Optimiser.cpp

```diff
@@ -93,6 +93,45 @@ namespace
 		BOOST_CHECK_EQUAL_COLLECTIONS(_expectation.begin(), _expectation.end(), output.begin(), output.end());
 	}
 
+	/// In contrast to the function `CSE`, this function doesn't finish the CSE optimization on an
```

Comment:
Since CSE is essentially KnownState + code regeneration and code regeneration does not work for verbatim data, maybe you could directly test KnownState instead?

Reply:
What would you test in KnownState? Currently, for `KnownState` from CSE, blocks would terminate at `verbatim`.

Reply:
But that's the point of one of the fixes I did: KnownState can actually process verbatim, but CSE cannot. KnownState is also used in different components like the gas meter.

Reply:
`verbatim` would still be fed into `KnownState` by CSE (during the call to `getOptimizedItems()`). So the other checks I could think about for only `KnownState` would be checking storage and memory contents. But that's too trivial, since they get emptied.

Reply:
I was thinking about feeding items into KnownState directly without the CSE and checking that the stack slot mechanism works as expected (consuming the arguments and replacing the return variables by fresh symbolic variables).

Reply:
I'll include such a test now.